core/runtime/src/text/encodings.rs

```diff
@@ -0,0 +1,87 @@
+pub(crate) mod utf8 {
+    use boa_engine::JsString;
+    use boa_engine::string::CodePoint;
+
+    pub(crate) fn encode(input: &JsString) -> Vec<u8> {
+        input
+            .code_points()
+            .flat_map(|s| match s {
+                CodePoint::Unicode(c) => c.to_string().as_bytes().to_vec(),
+                CodePoint::UnpairedSurrogate(_) => "\u{FFFD}".as_bytes().to_vec(),
+            })
+            .collect()
+    }
+
+    pub(crate) fn decode(input: &[u8]) -> JsString {
+        let string = String::from_utf8_lossy(input);
+        JsString::from(string.as_ref())
+    }
+}
+
+pub(crate) mod utf16le {
+    use boa_engine::{JsString, js_string};
+
+    pub(crate) fn encode(input: &JsString) -> Vec<u8> {
+        let bytes = input.as_str().to_vec();
+        let bytes = bytes.as_slice();
+        bytemuck::cast_slice(bytes).to_vec()
```

Comment:
Suggestion: I would be more efficient to call `.variant()` on `JsStr`, and match on the latin1 and utf16. That way we don't do any allocations if not required.